### PR TITLE
#39: improving Javadoc for UnitConverter#isLinear and #isIdentity

### PR DESCRIPTION
**Pull request by andi-huber**
This change is [Reviewable](https://reviewable.io/reviews/unitsofmeasurement/unit-api/182)

### CODE REVIEW

File: src/main/java/javax/measure/UnitConverter.java [hunk 1]
```diff
@@ -53,35 +54,50 @@
 
     /**
      * Indicates if this converter is an identity converter. The identity converter returns its input argument ({@code convert(x) == x}).
+     * <p>
+     * Note: Identity converters are also always 'linear'. {@link UnitConverter#isLinear()}
```
Comment (desruisseaux) [inline src/main/java/javax/measure/UnitConverter.java]: Was the lonely `{@link UnitConverter#isLinear()}` intended to be a `@see UnitConverter#isLinear()` (after `@return` tag)?

File: src/main/java/javax/measure/UnitConverter.java [hunk 2]
```diff
@@ -53,35 +54,50 @@
...
+     * for 'A' and 'B' the order of their composition does not matter. Expressed as Java code:
+     * </p>
      * <p>
-     * For linear converters the following property always hold:
+     *{@code A.concatenate(B).convert(u) == B.concatenate(A).convert(u)}
+     * </p>
+     * <p>
+     * Note: For composing UnitConverters see also {@link UnitConverter#concatenate(UnitConverter)}.
      * </p>
      *
```
Comment (desruisseaux) [inline src/main/java/javax/measure/UnitConverter.java]: Should we add "Note that <cite>linear transformation</cite> are not necessarily the same as <cite>linear function</cite>."? The fact that I interpreted this method as "is linear function", that is as an equation of the form "y = a*x + b" was the reason for my initial objection against the method name (the interpretation as "linear map" instead resolve that objection).

File: src/main/java/javax/measure/UnitConverter.java (hunk 1 above)
Comment (andi-huber) [reply src/main/java/javax/measure/UnitConverter.java]: Not sure, the intend is to provide more insight on the note itself (particularly the word 'linear') at the very same line.

Comment (desruisseaux) [reply src/main/java/javax/measure/UnitConverter.java]: Then what about the following? (note the replacement of `@link` by `@linkplain`)

```
Note: Identity converters are also always {@linkplain UnitConverter#isLinear() 'linear'}.
```

File: src/main/java/javax/measure/UnitConverter.java (hunk 2 above)
Comment (andi-huber) [reply src/main/java/javax/measure/UnitConverter.java]: Do you feel, that the proposed javadoc does still leave any room for misinterpretation?

File: src/main/java/javax/measure/UnitConverter.java (hunk 1 above)
Comment (andi-huber) [reply src/main/java/javax/measure/UnitConverter.java]: fine with me!

Comment (andi-huber) [reply src/main/java/javax/measure/UnitConverter.java]: ... same thing also applies to a later note below

File: src/main/java/javax/measure/UnitConverter.java (hunk 2 above)
Comment (desruisseaux) [reply src/main/java/javax/measure/UnitConverter.java]: I think the proposed javadoc is clear, To be frank, I was not aware that "linear map" and "linear function" could differ before this link to wikipedia, and my lack of knowledge about this difference caused me to react against the `isLinear` method name. I though that some other peoples may fall in the same trap, but I have no strong opinion - it is fine for me to leave the javadoc as is.

File: src/main/java/javax/measure/UnitConverter.java (hunk 1 above)
Comment (andi-huber) [reply src/main/java/javax/measure/UnitConverter.java]: I've updated this PR, but did not quite change the link as you suggest, but hope that suffices:
`Note: Identity converters are also always 'linear', see {@link UnitConverter#isLinear()}.`

File: src/main/java/javax/measure/UnitConverter.java (hunk 2 above)
Comment (andi-huber) [reply src/main/java/javax/measure/UnitConverter.java]: Ok, thx. No worries, I was confused as well. But I think our definition is bulletproof, that's what matters here.

File: src/main/java/javax/measure/UnitConverter.java (hunk 1 above)
Comment (desruisseaux) [reply src/main/java/javax/measure/UnitConverter.java]: Cool, thanks.

Comment (keilw) [reply src/main/java/javax/measure/UnitConverter.java]: Could it be merged now?

**Pull request merged**